src/nvim/ex_cmds.c

```diff
@@ -1056,6 +1058,20 @@ void do_bang(int addr_count, exarg_T *eap, bool forceit, bool do_in, bool do_out
     STRCAT(newcmd, prevcmd);
     STRCAT(newcmd, p_shq);
     free_newcmd = true;
+  } else {
+    if (addr_count != 0) {
```

Comment:
seems like a strange place for this to live. there's already a `if (addr_count == 0) {` check on line 1060/1076 below, and it has an `else{}` block.

Reply:
Thanks for reviewing this, I moved this part below as you suggested. However, I did that first because (again if I understood properly), in case the shell needs quote around the command, `newcmd` is set, and then may be executed as a filter program. I did that because the logic appeared simpler for me (I am sorry for the bad idea). However, in the current state of the code, shells that needs quotes are then not managed properly. I do not have an example for such a type of shells, and a web search did not return something. Do you, in any case, have an idea of a such a shell to test the code ? I thank you for your time.